internal/handlers/pg/pgdb/indexes.go

```diff
@@ -16,49 +16,118 @@ package pgdb
 )
 
-// indexParams contains parameters for creating an index.
-// TODO This type will become exported in https://github.com/FerretDB/FerretDB/issues/1509 (similar to QueryParams).
-type indexParams struct {
-	db         string   // FerretDB database name
-	collection string   // FerretDB collection name
-	index      string   // FerretDB index name
-	key        indexKey // Index specification (pairs of field names and sort orders) // TODO
-	unique     bool     // Whether the index is unique
+// IndexParams contains parameters for creating an index.
+type IndexParams struct {
+	Index  string   // FerretDB index name
+	Key    IndexKey // Index specification (pairs of field names and sort orders) // TODO
+	Unique bool     // Whether the index is unique
 }
 
-// indexKey defines a type for index key - pairs of field names and sort orders.
-type indexKey []indexKeyPair
+// IndexKey defines a type for index key - pairs of field names and sort orders.
+type IndexKey []IndexKeyPair
 
-// indexKeyPair consists of a field name and a sort order that are part of the index.
-type indexKeyPair struct {
-	field string
-	order indexOrder
+// IndexKeyPair consists of a field name and a sort order that are part of the index.
+type IndexKeyPair struct {
+	Field string
+	Order IndexOrder
 }
 
-// indexOrder defines a type for index sort order.
-type indexOrder int8
+// IndexOrder defines a type for index sort order.
+type IndexOrder int8
 
-// indexOrder constants.
+// IndexOrder constants.
 const (
-	indexOrderAsc  indexOrder = 1
-	indexOrderDesc indexOrder = -1
+	IndexOrderAsc  IndexOrder = 1
+	IndexOrderDesc IndexOrder = -1
 )
 
+// Indexes returns a list of indexes for the given database and collection.
+//
+// If the given collection does not exist, it returns ErrTableNotExist.
+func Indexes(ctx context.Context, tx pgx.Tx, db, collection string) ([]IndexParams, error) {
+	metadata, err := newMetadata(tx, db, collection).get(ctx, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if !metadata.Has("indexes") {
+		return []IndexParams{}, nil
+	}
+
+	indexes := must.NotFail(metadata.Get("indexes")).(*types.Array)
+
+	res := make([]IndexParams, indexes.Len())
+	iter := indexes.Iterator()
+
+	defer iter.Close()
+
+	for {
+		i, idx, err := iter.Next()
+
+		switch {
+		case err == nil:
+			idx := idx.(*types.Document)
+			key := must.NotFail(idx.Get("key")).(*types.Document)
+
+			res[i] = IndexParams{
+				Index:  must.NotFail(idx.Get("name")).(string),
+				Unique: must.NotFail(idx.Get("unique")).(bool),
+				Key:    make([]IndexKeyPair, 0, key.Len()),
+			}
+
+			keyIter := key.Iterator()
+
+			for i := 0; i < key.Len(); i++ {
+				var field string
+				var value any
+				field, value, err = keyIter.Next()
+
+				switch {
+				case err == nil:
+					res[i].Key = append(res[i].Key, IndexKeyPair{
+						Field: field,
+						Order: IndexOrder(value.(int32)),
+					})
+				case errors.Is(err, iterator.ErrIteratorDone):
```

Comment:
```suggestion
				case errors.Is(err, keyIter.ErrIteratorDone):
```

Reply:
Indeed, I think I misread something totally.. Ignore my comment.

Reply:
By the way, why is this line not covered by the test? codecov thinks so.

Reply:
Because we exit the loop before we come here:

```
for i := 0; i < key.Len(); i++
```

And I need `i` to address `res[i]`. But I don't like this duplication of iteration... Do you have an idea how to improve it?

Reply:
Maybe I can just ignore this error actually :D

Reply:
fixed